FireboltPrivacy/FireboltPrivacyImplementation.cpp

```diff
@@ -0,0 +1,138 @@
+/*
+ * If not stated otherwise in this file or this component's LICENSE file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2020 Metrological
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Module.h"
+#include <interfaces/IConfiguration.h>
+#include <interfaces/IFireboltPrivacy.h>
+
+namespace WPEFramework {
+
+namespace Plugin
+{
+    class FireboltPrivacyImplementation : public Exchange::IFireboltPrivacy,
+                                         public Exchange::IConfiguration {
+
+    private:
+        class Config : public Core::JSON::Container {
+        public:
+            Config(const Config&) = delete;
+            Config& operator=(const Config&) = delete;
+
+            Config()
+                : Core::JSON::Container()
+                , devicemanifest("")
+                , fireboltconfigpath("")
+            {
+                Add(_T("devicemanifest"), &devicemanifest);
+                Add(_T("fireboltconfigpath"), &fireboltconfigpath);
+            }
+            ~Config() override = default;
+        public:
+            Core::JSON::String devicemanifest;
+            // This path is specific to firebolt. It points to the path where
+            // firebolt config files are placed. These files are shared between Ripple and other FireboltThunderPlugins
+            // In prod it points to /etc/
+            // and in dev it points to /opt/
+            Core::JSON::String fireboltconfigpath;
```

Comment:
Perhaps this requires some architectural design. I certainly would not give all Ripple plugins this option as when  it changes you need to change the config of all plugins. Perhaps use an environment variable to specify the ripple config root or perhaps use a ripple code library that holds a lot of default functionality for Ripple (like we have our ThunderLibaries)? Then you can have consistency for all Ripple plugins.
But perhaps another question why would you need to access current Ripple configs, would that not be covered in a normal Thunder Plugin? Why would there be a ripple ecosystem and a Thunder ecosystem? If so I guess we would advice to move everything to Thunder plugins where every plugin has its own responsibility and if plugins need to communicate then do that via a strict defines abstract (comrpc) interface.